compiler/passes/codegen.cpp

```diff
@@ -530,6 +530,67 @@ static void codegen_header_compilation_config() {
 }
 
 
+static void protectNameFromC(Symbol* sym) {
+  //
+  // Let's assume we only have to rename user symbols and that we've
+  // done a good job of protecting our internal and standard module
+  // symbols.  This has the advantage of not having to take special
+  // pains to keep from renaming things like chpl_string and uint64_t
+  // (which perhaps should arguably also have FLAG_EXTERN, but don't
+  // consistently seem to today; and adding FLAG_EXTERN to them seems
+  // to result in it getting propagated to type aliases and such in
+  // ways that caused headaches for me).
+  //
```

Comment:
That's a good point.  Initially I was doing them all, but running into challenges (strangely, not with things like int64_t themselves once I marked them as FLAG_EXTERN, but rather with typedefs that aliased that type).  To dodge that issue, I'd convinced myself that our code didn't need to be protected because once it was working in a conflict-free way, we'd know it's safe; but as you point out, that assumes that we know all libraries/symbols against which we'll be linking (all environments in which users will use our code).  So I think it's right for me to push harder on that case.  Would you like me to do it before committing, or can we consider this a positive, partial step forward worth putting in and then working on it independently?  (either is OK with me).